jws/jws.go

```diff
@@ -363,7 +363,7 @@ func VerifyWithJWKSet(buf []byte, keyset *jwk.Set, keyaccept JWKAcceptFunc) (pay
 			continue
 		}
 
-		payload, err := VerifyWithJWK(buf, key)
+		payload, err = VerifyWithJWK(buf, key)
```

Comment:
It's interesting that the tests pass with and without this change. Could you add a test as part of this fix?